client/src/views/FlightsAll.vue

```diff
@@ -35,8 +32,13 @@ const props = defineProps({
 });
 const route = useRoute();
 
-const { fetchFlights, filterActive, clearFilter, isLoading } = useFlights();
-await fetchFlights({ params: route.params, queries: route.query });
+const { fetchData, setApiEndpoint, setPaginationSupported } =
+  useData("flights");
+
+setApiEndpoint(ApiService.getFlights);
```

Comment:
wenn ich das setApiEndpoint nicht setze funktioniert das fetchdata nicht.
irgendwie finde ich das nicht so sexy wie man useData benutzen soll.

> Composables = Transparent Source of Data and Functions

https://vueschool.io/articles/vuejs-tutorials/what-is-a-vue-js-composable/


Dann wäre es doch besser wenn das label ein api Endpunkt wäre. dann wäre useData genauso wiederverwendbar und es bräuchte nicht dieses umständliche setApiEndpoint.

So wie hier zu sehen: https://github.com/vueschool/vue-3-composition-api/tree/main/src/composables

Reply:
Die Idee mit dem Endpoint als Label finde ich gut. Werde ich umsetzen.